docker/Dockerfile

```diff
@@ -0,0 +1,79 @@
+FROM ubuntu:18.04
+MAINTAINER Pramod Kumbhar <pramod.s.kumbhar@gmail.com>
+MAINTAINER Matthias Wolf <matthias.wolf@epfl.ch>
+
+ENV DEBIAN_FRONTEND noninteractive
+
+ARG proxy
+ENV http_proxy=$proxy
+ENV https_proxy=$proxy
+
+# default software required
+RUN apt-get update
+RUN apt-get -y install software-properties-common
+RUN add-apt-repository ppa:ubuntu-toolchain-r/test
+RUN apt-get update
+RUN apt-get -y \
+            --allow-downgrades \
+            --allow-remove-essential \
+            --allow-change-held-packages \
+            install \
+                autoconf \
+                automake \
+                bison \
+                build-essential \
+                cmake \
+                curl \
+                cython \
+                diffutils \
+                flex \
+                gfortran \
+                git \
+                libboost-all-dev \
+                libbz2-dev \
+                libhdf5-serial-dev \
+                libncurses-dev \
+                libpciaccess-dev \
+                libopenmpi-dev \
+                libreadline-dev \
+                libtool \
+                libxml2-dev \
+                make \
+                numactl \
+                pkg-config \
+                python2.7-dev \
+                python-minimal \
+                python-numpy \
+                software-properties-common \
+                tcl-dev \
+                unzip \
+                vim \
+                zlib1g-dev
+
+RUN git clone https://github.com/BlueBrain/spack.git
+ENV SPACK_ROOT $PWD/spack
+ENV PATH $SPACK_ROOT/bin:$PATH
+
+# see this: http://stackoverflow.com/questions/20635472/using-the-run-instruction-in-a-dockerfile-with-source-does-not-work
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
+RUN . $SPACK_ROOT/share/spack/setup-env.sh
```

Comment:
Line 59 can be removed right?

Reply:
Sourcing the environment? Don't think so. Re-aliasing the shell could be removed, though.

Reply:
`source` only applies to the `RUN` instruction. Environment is lost afterward.

Reply:
Like Jenkins… Thanks! I hoped that Docker would be "smarter"